cloudify_deployment_proxy/tasks.py

```diff
@@ -12,25 +12,82 @@
 #    * See the License for the specific language governing permissions and
 #    * limitations under the License.
 
+import sys
+
+from cloudify import ctx
 from cloudify.decorators import operation
+from cloudify.exceptions import NonRecoverableError
+from cloudify.utils import exception_to_error_cause
+
+
 from . import DeploymentProxyBase
 
 
+def generate_traceback_exception():
+    _, exc_value, exc_traceback = sys.exc_info()
+    response = exception_to_error_cause(exc_value, exc_traceback)
+    return response
+
+
 @operation
 def upload_blueprint(**_):
-    return DeploymentProxyBase(_).upload_blueprint()
+    try:
```

Comment:
(MINOR) Maybe is it possible to implement try / except block in generic manner - e.g. as decorator or new method taking method (object) to be called under try as argument